src/strings/nfa-strings.cc

```diff
@@ -198,3 +198,16 @@ std::set<std::pair<int, int>> Mata::Strings::get_word_lengths(const Nfa::Nfa& au
 
     return ret;
 }
+
+bool Mata::Strings::is_lang_eps(const Nfa::Nfa& aut) {
+    Nfa::Nfa tr_aut = aut.get_trimmed_automaton();
+    if(tr_aut.initial.size() == 0)
+        return false;
+    for(const auto& ini : tr_aut.initial) {
+        if(!tr_aut.final[ini])
+            return false;
+        if(tr_aut.delta[ini].size() > 0)
```

Comment:
```suggestion
        if(!tr_aut.delta[ini].empty())
```

Reply:
well, these things are more like a matter of taste

Reply:
That is true. I believe that testing for emptiness with `empty()` is more idiomatic. However, if we change the underlying container (with a user-defined one even),  `size()` could have linear time complexity, while as `empty()` will be always a constant time operation.

Reply:
if we introduce `size` as a linear operation at some point, we deserve nothing but shame and bad performance

Reply:
Looks like we finally have Martin's successor in this project 8-)

Reply:
Well, for some tree-like structures or linked lists, the `size()` operation could truly be linear.

Nevertheless, I am OK with keeping the current implementation as well.

@tfiedor Says the man pointing out for loop condition constants extraction :) Anyway, could you review the PR so that if you find nothing wrong with it, we can merge the changes now and can use the introduced function in Noodler first thing in the morning?

Reply:
Sorry. You were not actually assigned to the PR as a reviewer. Feel free to ignore my previous message. I will merge the PR later today, if no one comes with anything new.

Reply:
> Anyway, could you review the PR so that if you find nothing wrong with it, we can merge the changes now and can use the introduced function in Noodler first thing in the morning?

 Sorry. You were not actually assigned to the PR as a reviewer. Feel free to ignore my previous message.

Reply:
@Adda0 I meant Vojta through this comment:
> if we introduce `size` as a linear operation at some point, we deserve nothing but shame and bad performance
This sounds like an old Martin's Wisdom 😎

Reply:
Now I can maybe see the similarities. Sorry for misunderstanding at first.

Reply:
Well, you are practically a successor of Martin's job :P

Reply:
> Looks like we finally have Martin's successor in this project 😎
❤️❤️❤️